homeassistant/components/mqtt/discovery.py

```diff
@@ -175,9 +182,25 @@ async def async_device_message_received(msg):
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
 
     hass.data[DISCOVERY_UNSUBSCRIBE] = await mqtt.async_subscribe(
-        hass, f"{discovery_topic}/#", async_device_message_received, 0
+        hass, f"{discovery_topic}/#", async_entity_message_received, 0
     )
     hass.data[LAST_DISCOVERY] = time.time()
+    mqtt_integrations = await async_get_mqtt(hass)
+
+    for (integration, topics) in mqtt_integrations.items():
+
+        async def async_integration_message_received(msg):
```

Comment:
Once this is fired once, should we unsubscribe the topic? Otherwise we will end up initializing a lot of flows.

Reply:
Maybe we can look at if we see an abort with reason `already_configured` and unsub then.

Reply:
OK, sounds good. Should we check for either of `single_instance_allowed` or `already_configured`?